### ISSUE TITLE
v4 breaks perl linter include paths

### ISSUE THREAD

**Issue by ralesk** [SublimeLinter/SublimeLinter#1252]
I'm really not sure where to report it but maybe this is a good place to start (and maybe it's just some weird configuration error, I don't know)

I use SL for linting Perl, and in my project the sources are all under `<projectroot>/src`.  In the v3 days, the linter would figure out the paths from the file's own position rather than what it seems to be doing now, basing it all at project root (as shown in the error about `@INC` in the Sublime Text status bar)

Thing is, however, that if I run my Perl program from another directory it will have a perfectly fine `@INC` and will run, so I'm not entirely sure what SublimeLinter is doing differently.

Here's an example program and a few screenshots:

![image](https://user-images.githubusercontent.com/1564502/37668348-8fd35f86-2c64-11e8-8e79-03a5aef92bf6.png)

With FindBin enabled it seems that the inc will contain .../sl-path-test/ and .../sl-path-test/lib, instead of the same with src in the middle, like it would during v3:

![image](https://user-images.githubusercontent.com/1564502/37668527-05ce2d88-2c65-11e8-9794-bfb082c77ca9.png)

Here's what actually running `perl -c` does:

Without FindBin, rightfully not finding SLPathTestModule:

```
~/Projects/sl-path-test$ perl -c src/test.pl
Can't locate SLPathTestModule.pm in @INC (you may need to install the SLPathTestModule module) (@INC contains: /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at src/test.pl line 15.
BEGIN failed--compilation aborted at src/test.pl line 15.
~/Projects/sl-path-test/src$ perl -c test.pl
syntax error at test.pl line 20, near "say"
test.pl had compilation errors.
```

With FindBin (as it allows for adding directories relative to the program being run, it shouldn't, and doesn't matter where I run `perl -c` from):

```
~/Projects/sl-path-test$ perl -c src/test.pl
syntax error at src/test.pl line 20, near "say"
src/test.pl had compilation errors.
~/Projects/sl-path-test/src$ perl -c test.pl    
syntax error at test.pl line 20, near "say"
test.pl had compilation errors.
```

In short: in the second case, SL should work too, but it doesn't.
It seems that in the v3 days SL's Perl linter plugin would be executing `perl -c` from the file's own directory, rather than from project root.

[sl-path-test.tar.gz](https://github.com/SublimeLinter/SublimeLinter/files/1830236/sl-path-test.tar.gz)
(set project root to sl-path-test, not to sl-path-test/src)

**Comment by kaste** [comment]
Yes, we changed the default working dir here. But you can set it in the linter settings to $file to get the old behavior

**Comment by ralesk** [comment]
Hmm, I don't see where I can do that though, the system-wide SublimeLinter.sublime-settings doesn't have a key (or a comment) that'd be about this.

**Comment by kaste** [comment]
http://www.sublimelinter.com/en/stable/linter_settings.html#working-dir

**Comment by ralesk** [comment]
*just* found it, like right this moment when you commented :D  Thanks though!

```
{
	"linters" : {
		"perl": {
			"working_dir": "$file_path"
		}
	}
}
```

Still rather odd that executing the same command from a shell (outside of `$file_path` that is) gives a different result than running SL, though.  (The FindBin case in my example)

**Comment by kaste** [comment]
I just ignored that bc it seemed impossible. 😁

**Comment by braver** [comment]
Can we close this?
